File: 05_arrays/05_08_linear_search/linear_search.cpp
#include <bits/stdc++.h>

using namespace std;

int linearSearch(vector<int> &nums, int key) {
    for (size_t i = 0; i < nums.size(); i++)
    {
        if (nums[i] == key)
        {
            return i;
        }
    }
    return -1;
}

void print(vector<int>& nums) {
    for (size_t i = 0; i < nums.size(); i++)
    {
        cout << nums[i] << " ";
    }
    cout << endl;
}

int main() {
    vector<int> nums = {1, 2, 3, 4, 5};
    int key = 3;

    cout << "Elements:-" << endl;
    print(nums);
    cout << "The elements " << key << " is at index: " << linearSearch(nums, key) << endl;
}

File: 05_arrays/05_08_linear_search/LinearSearch.java
import java.util.ArrayList;

public class LinearSearch {
    public static void main(String[] args) {
        ArrayList<Integer> nums = new ArrayList<>();
        nums.add(1);
        nums.add(2);
        nums.add(3);
        nums.add(4);
        nums.add(5);

        int key = 3;

        System.out.println("Elements:-");
        print(nums);
        System.out.println("The element " + key + " is at index: " + linearSearch(nums, key));
    }

    public static int linearSearch(ArrayList<Integer> nums, int key) {
        for (int i = 0; i < nums.size(); i++) {
            if (nums.get(i) == key) {
                return i;
            }
        }
        return -1;
    }

    public static void print(ArrayList<Integer> nums) {
        for (int num : nums) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
